### PR TITLE
Version/v1.1.0

### PR DESCRIPTION
**Pull request by objmj-itminds**
Introducing the magical directory of "src".
Now instead of all our code being in the root dir it now resides in the src dir.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -34,8 +36,20 @@ Array.prototype.sum = function (func = x => x, thisArg = this) {
...
 
-Array.prototype.average = function (func = x => x, thisArg = this) {
+Array.prototype.average = function (
+  func = x => x,
+  round = null,
+  thisArg = this
+) {
   if (thisArg.length <= 0) throw Error("Out of bounds");
+  if (round !== null) {
+    if (round <= 0) throw Error("Out of bounds");
```
Comment (phillip-itminds) [inline src/index.ts:46]: Couldn't `round == 0` be allowed? It could be taken as the same case as `round == null`.

File: src/index.ts [hunk 2]
```diff
@@ -50,9 +64,7 @@ Array.prototype.max = function (func = x => x, thisArg = this) {
 Array.prototype.shuffle = function (thisArg = this) {
   return thisArg
     .map(val => ({
-      sort:
-        Math.floor(Math.pow(10, 14) * Math.random() * Math.random()) %
-        (thisArg.length * 100),
+      sort: Math.floor(1e14 * Math.random() ** 2) % (thisArg.length * 100),
```
Comment (phillip-itminds) [inline src/index.ts:67]: Why this exact formula? It will give a random value, though, so all's fine. Just curious.

File: tests/index.test.ts [hunk 3]
```diff
@@ -1,154 +1,176 @@
...
+    let result = myArr.sortByAttr(a => a.age, 1);
 
-    expect(result[0].score).equal(1000);
-    expect(result[4].score).equal(150);
+    expect(result[0].age).equal(ASSERTIONS.MAX_AGE);
+    expect(result[myArr.length - 1].age).equal(ASSERTIONS.MIN_AGE);
   });
 
   it("ASC by string with num", function () {
-    let result = myArr.sortByAttr(a => a.nickName);
```
Comment (phillip-itminds) [inline tests/index.test.ts:50]: You don't use the `num` here as you do in the tests above.

File: tests/index.test.ts [hunk 4]
```diff
@@ -1,154 +1,176 @@
...
     let roll = 0;
-    let counts: { [k: number]: number } = {};
+    let counts: { [k: string]: number } = {};
 
-    for (let i = 1; i <= 5; i++) {
+    for (let i = 0; i < myArr.length; i++) {
       counts[i] = 0;
     }
 
     while (roll < 1000000) {
```
Comment (phillip-itminds) [inline tests/index.test.ts:105]: Perhaps write a comment about how you actually test this. I'm not really following your test here.

**Review by phillip-itminds — COMMENTED** [review]
There are some minor things that could be improved, but it looks fine overall.

**Review by phillip-itminds — APPROVED** [review]

**Review by rbs-itminds — APPROVED** [review]

**Pull request merged**